bedrock/mozorg/templatetags/misc.py

```diff
@@ -628,6 +628,20 @@ def play_store_url(ctx, product, campaign=None):
     return base_url
 
 
+@library.global_function
+@jinja2.pass_context
+def ms_store_url(ctx, product, campaign=None, mode="direct"):
+    """Returns a Microsoft Windows Store URL for a given product"""
+    base_url = getattr(settings, f"MICROSOFT_WINDOWS_STORE_{product.upper()}_LINK")
+    params = f"?mode={mode}"
+    base_url = base_url + params
+
+    if campaign:
+        base_url = base_url + f"&cid={campaign}"
+
+    return base_url
```

Comment:
We have the `urlparams` helper that can make some of this logic a bit easier...

```python
from bedrock.base.templatetags.helpers import urlparams

@library.global_function
@jinja2.pass_context
def ms_store_url(ctx, product, campaign=None, mode="direct"):
    """Returns a Microsoft Windows Store URL for a given product"""
    base_url = getattr(settings, f"MICROSOFT_WINDOWS_STORE_{product.upper()}_LINK")
    params = {
        "mode": mode,
        "cid": campaign,  # If None, it will not be in the URL.
    }

    return urlparams(base_url, **params)
```